nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/UnpackContent.java

```diff
@@ -154,75 +171,85 @@ protected void init(final ProcessorInitializationContext context) {
-        FlowFile flowFile = session.get();
-        if (flowFile == null) {
-            return;
-        }
+    @OnStopped
+    public void onStopped() {
+        unpacker = null;
+        fileFilter = null;
+    }
 
-        final ProcessorLog logger = getLogger();
-        String packagingFormat = context.getProperty(PACKAGING_FORMAT).getValue().toLowerCase();
-        if (AUTO_DETECT_FORMAT.equals(packagingFormat)) {
-            final String mimeType = flowFile.getAttribute(CoreAttributes.MIME_TYPE.key());
-            if (mimeType == null) {
-                logger.error("No mime.type attribute set for {}; routing to failure", new Object[]{flowFile});
-                session.transfer(flowFile, REL_FAILURE);
-                return;
-            }
+    @OnScheduled
+    public void onScheduled(ProcessContext context) throws ProcessException {
+        if (fileFilter == null) {
+            fileFilter = Pattern.compile(context.getProperty(FILE_FILTER).getValue());
+            tarUnpacker = new TarUnpacker(fileFilter);
+            zipUnpacker = new ZipUnpacker(fileFilter);
+            flowFileStreamV3Unpacker = new FlowFileStreamUnpacker(new FlowFileUnpackagerV3());
+            flowFileStreamV2Unpacker = new FlowFileStreamUnpacker(new FlowFileUnpackagerV2());
+            flowFileTarUnpacker = new FlowFileStreamUnpacker(new FlowFileUnpackagerV1());
+        }
 
-            switch (mimeType.toLowerCase()) {
-                case "application/tar":
-                    packagingFormat = TAR_FORMAT;
-                    break;
-                case "application/x-tar":
-                    packagingFormat = TAR_FORMAT;
-                    break;
-                case "application/zip":
-                    packagingFormat = ZIP_FORMAT;
-                    break;
-                case "application/flowfile-v3":
-                    packagingFormat = FLOWFILE_STREAM_FORMAT_V3;
-                    break;
-                case "application/flowfile-v2":
-                    packagingFormat = FLOWFILE_STREAM_FORMAT_V2;
-                    break;
-                case "application/flowfile-v1":
-                    packagingFormat = FLOWFILE_TAR_FORMAT;
-                    break;
-                default: {
-                    logger.info("Cannot unpack {} because its mime.type attribute is set to '{}', which is not a format that can be unpacked; routing to 'success'", new Object[]{flowFile, mimeType});
-                    session.transfer(flowFile, REL_SUCCESS);
-                    return;
-                }
-            }
+        PackageFormat format = PackageFormat.getFormat(context.getProperty(PACKAGING_FORMAT).getValue());
+        if (format != PackageFormat.AUTO_DETECT_FORMAT && unpacker == null) {
+            initUnpacker(format);
         }
+    }
 
-        final Unpacker unpacker;
-        final boolean addFragmentAttrs;
+    public void initUnpacker(PackageFormat packagingFormat) {
         switch (packagingFormat) {
```

Comment:
This one complains with the compile-warning "The enum constant AUTO_DETECT_FORMAT needs a corresponding case label in this enum switch on UnpackContent.PackageFormat". Can you take a look?